internal/controllers/historicalv2.go

```diff
@@ -0,0 +1,101 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/DIMO-Network/device-data-api/internal/services/elastic"
+	"github.com/DIMO-Network/shared/middleware/privilegetoken"
+	"github.com/gofiber/fiber/v2"
+)
+
+// GetHistoricalPermissionedV2 godoc
+// @Description  Get all historical data for a tokenID, within start and end range
+// @Tags         device-data
+// @Produce      json
+// @Success      200
+// @Param        tokenID  path   int64  true   "token id"
+// @Param        startTime     query  string  false  "startTime is an RFC3339 formatted date-time string. if empty two weeks back"
+// @Param        endTime       query  string  false  "endTime is an RFC3339 formatted date-time string. if empty today"
+// @Param        buckets       query  string  false  "number of data points to return, default 1000"
+// @Security     BearerAuth
+// @Router       /v2/vehicle/{tokenID}/history [get]
+func (d *DeviceDataController) GetHistoricalPermissionedV2(c *fiber.Ctx) error {
+	// get the startTime and endTime from the query parameters and convert them to time.Time
+	var err error
+	startTime, err := getTimeFromQuery(c, "startTime")
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, err.Error())
+	}
+	endTime, err := getTimeFromQuery(c, "endTime")
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, err.Error())
+	}
+
+	buckets, err := strconv.ParseInt(c.Query("buckets", "0"), 10, 64)
```

Comment:
Do we want to cap this? Couldn't it get out of hand?

Reply:
What if it's negative?

Reply:
> Do we want to cap this? Couldn't it get out of hand?

The natural cap is the number of ms between start and stop time. Our other cap would be the context, so 5 minutes, which is probably too long. We can either give it a shorter timeout or an arbitrary size cap.

> What if it's negative?

Negative cause weird behavior with the interval calculation I am going to treat `buckets <= 0` as invalid.

Reply:
Letting the timeout cap it is a bit spooky to me but I'm fine to leave it.